livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -168,18 +168,16 @@ private String extractStatementResult(String result) {
   @Override
   public void cancel(InterpreterContext context) {
     if (livyVersion.isCancelSupported()) {
+      String paraId = context.getParagraphId();
```

Comment:
paraId would not be null since it is got from `InterpreterContext`, unless it is unit test that we didn't construct `InterpreterContext` properly. But I don't think we need to do any special check for unit test.